javaslang/src/main/java/javaslang/collection/List.java

```diff
@@ -85,9 +81,80 @@
  * @author Daniel Dietrich
  * @since 1.1.0
  */
-public interface List<T> extends Kind1<List<?>, T>, LinearSeq<T>, Stack<T> {
+public final class List<T> implements Stack<T>, LinearSeq<T>, Kind1<List<?>, T>, Serializable {
```

Comment:
@paplorinc Ohhh my gosh - I just saw that you made List a class O_O. You can't do such big changes in a PR named "Added ... Benchmarks". List is an algebraic data type, like Option (Some/None). We need to distinguish the states by type. It is essential for pattern matching. Also we follow best practices from the book "Purely Functional Data Structures". Scala and Haskell do the same for example.

Please revert the List changes and open a discussion/issue if you think there is potential to enhance the implementation.